metadata/src/main/java/org/apache/kafka/metadata/bootstrap/BootstrapMetadata.java

```diff
@@ -149,6 +149,32 @@ public BootstrapMetadata copyWithOnlyVersion() {
                 metadataVersion, source);
     }
 
+    public BootstrapMetadata copyWithFeatureRecord(String featureName, short level) {
+        List<ApiMessageAndVersion> newRecords = new ArrayList<>();
+        int i = 0;
+        while (i < records.size()) {
+            if (records.get(i).message() instanceof FeatureLevelRecord) {
+                FeatureLevelRecord record = (FeatureLevelRecord) records.get(i).message();
+                if (record.name().equals(featureName)) {
+                    FeatureLevelRecord newRecord = record.duplicate();
+                    newRecord.setFeatureLevel(level);
+                    newRecords.add(new ApiMessageAndVersion(newRecord, (short) 0));
+                    break;
```

Comment:
Do we need a comment for this method?
It seems that this is a workaround for a particular scenario. Otherwise, there is no reason we should break in a "copy" method.

Reply:
@frankvicky Could you please open a jira for it?